README.md

```diff
@@ -94,17 +95,41 @@ class BundledMailNotification extends Notification implements ShouldBundleNotifi
 }
 ```
 
+## Limitations
+
 Because of limitations in Laravel, the database channel must implicitly use the `toArray`, or `toDatabase` method.
 To get the notifications in those functions, you can use the `getBundle()` method.
 
 ```php
-public function toDatabase(object $notifiable)
+public function toDatabase(object $notifiable): array
 {
     $notifications = $this->getBundle();
     return ['names' => $notifications->pluck('name')->toArray()];
 }
 ```
 
+When you want to add custom middleware, it is important to always apply the bundle middleware first.
+If you don't do this, your notification could be bundled with a another notification later on, which can cause unexpected results.
```

Comment:
Would be cool if we could have a similar solution as eloquent traits, where you are able to add a `boot[trait name]` method to your trait.